GPSlog_Basic/app/src/main/java/com/example/gpslog/Segment.java

```diff
@@ -53,128 +63,149 @@ public Segment(ArrayList<Track> tracks){
+    private long trackTimeDiff(Track startTrack, Track endTrack) throws ParseException {
+        Date timeStart = df.parse(startTrack.time);
+        Date timeEnd = df.parse(endTrack.time);
+        return (timeEnd.getTime() - timeStart.getTime());
     }
 
-    // Returns the time length of the Segment
-    public long getSegTimeLength() {
-        return segTimeLength;
-    }
+    // Returns the distance traveled between two tracks in meters
+    private double trackDist(Track startTrack, Track endTrack){
+        Location startLoc = new Location("");
+        startLoc.setLatitude(startTrack.latitude);
+        startLoc.setLongitude(startTrack.longitude);
+        Location lastLoc = new Location("");
+        lastLoc.setLatitude(endTrack.latitude);
+        lastLoc.setLongitude(endTrack.longitude);
 
-    // Sets the time length of the Segment to be the long passed in
-    public void setSegTimeLength(long segTimeLength) {
-        this.segTimeLength = segTimeLength;
+        return lastLoc.distanceTo(startLoc);
     }
 
-    // Sets the time length of the Segment from the ArrayList of tracks passed in
-    private void setSegTimeLength(ArrayList<Track> tracks) throws ParseException {
-        Date time1 = new Date();
-        Date time2 = new Date();
-        time1 = df.parse(tracks.get(0).time);
-        time2 = df.parse(tracks.get(tracks.size() - 1).time);
-        segTimeLength = time2.getTime() - time1.getTime();
+    // Returns the stops with respect to time of the Segment
+    public double getStopsWRTime() {
+        return stopsWRTime;
     }
 
-    // Returns the distance traveled in meters
-    public double getDistanceTraveled() {
-        return distanceTraveled;
+    // Sets the stops WR time to be the double passed in
+    public void setStopsWRTime(int stops, long time) {
+        if (stops != 0 && time != 0) {
+            this.stopsWRTime = stops / time;
+        }
+        else {
+            this.stopsWRTime = 0.0;
+        }
     }
 
-    // Sets the distance traveled to be the double passed in
-    public void setDistanceTraveled(double distanceTraveled) {
-        this.distanceTraveled = distanceTraveled;
+    // Returns the stops with respect to distance of the Segment
+    public double getStopsWRSpace() {
+        return stopsWRSpace;
     }
 
-    // Sets the distance traveled from the ArrayList of tracks passed in
-    private void setDistanceTraveled(ArrayList<Track> tracks){
-        Location startLoc = new Location("");
-        Track firstTrack = tracks.get(0);
-        startLoc.setLatitude(firstTrack.latitude);
-        startLoc.setLongitude(firstTrack.longitude);
-        Location lastLoc = new Location("");
-        Track lastTrack = tracks.get(tracks.size() - 1);
-        lastLoc.setLatitude(lastTrack.latitude);
-        lastLoc.setLongitude(lastTrack.longitude);
-
-        distanceTraveled = lastLoc.distanceTo(startLoc);
+    // Sets the stops WR distance to be the double passed in
```

Comment:
This seems kinda misleading considering that stopsWRSpace is not just getting set. Both values are used in the fraction.